sympy/solvers/ode.py

```diff
@@ -1218,8 +1218,7 @@ class in it.  Note that a hint may do this anyway if
                 matching_hints[s + "_Integral"] = r
                 if undetcoeff['test']:
                     r['trialset'] = undetcoeff['trialset']
-                    matching_hints["nth_linear_constant_coeff_undetermined_"
-                        "coefficients"] = r
+                    matching_hints["nth_linear_constant_coeff_undetermined_coefficients"] = r
```

Comment:
to me, the 80 char limit is the most breakable of style guidelines, but I don't feel that strongly about it, so if others insist, I can make that change easily.